lib/fixed-points.nix

```
@@ -394,4 +394,77 @@ rec {
     fix' (self: (rattrs self) // {
       ${extenderName} = f: makeExtensibleWithCustomName extenderName (extends f rattrs);
     });
+
+  /**
+    Convert to an extending function (overlay).
+
+    `toExtension` is the `toFunction` for extending functions (a.k.a. extensions or overlays).
+    It converts a non-function or a single-argument function to an extending function,
+    while returning a double-argument function as-is.
+
+    That is, it takes one of `x`, `prev: x`, or `final: prev: x`,
+    and returns `final: prev: x`, where `x` is not a function.
+
+    This function is extracted from the implementation of
+    the fixed-point arguments support of `stdenv.mkDerivation`.
+    It bridges the gap between `<pkg>.overrideAttrs`
+    before and after the overlay-style support,
+    as well as `config.packageOverrides` and `config.overlays` in `pkgs`.
+
+    # Inputs
+
+    `f`
+    : The function or non-function to convert to an extending function.
```

```suggestion
    : The function or value to convert to an extending function.
```

This looks a little less awkward IMO.

It could be further simplified to "the value", but then we're losing a good hint that it's often a function, so I think "function or value" is best, despite functions being values.